lib/warder/arguments.rb

```diff
@@ -0,0 +1,104 @@
+require 'optparse'
+
+module Warder
+  # responsible for parsing cli arguments
+  class Arguments
+    def initialize(argv, stdout, kernel)
+      @argv = argv
+      @stdout = stdout
+      @kernel = kernel
+      @options = {}
+    end
+
+    def parse
+      parse_options
+      assign_files
+      OpenStruct.new(@options)
+    end
+
+    private
+
+    def parse_options
+      OptionParser.new do |opts|
+        quiet(opts)
+        stats(opts)
+        combined(opts)
+        validators(opts)
+      end.parse!(@argv)
+    end
+
+    def assign_files
+      @options['files'] = @argv.empty? ? '.' : @argv.join(' ')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.